mbq/atomiq/management/commands/atomic_run_consumer.py

```diff
@@ -79,6 +79,17 @@ def collect_task_metrics(self, queue, task, execution_start, execution_end):
             'queue_type': queue,
         }
 
+        # `task` can be a `SNSTask`, `SQSTask`, or `CeleryTask`; duck type `task` to determine
```

Comment:
is duck typing really the only way? :-/

Reply:
No, not the only way; we could use `isinstance`, for example, but [I tend to not use it](http://canonical.org/~kragen/isinstance/), even though that's just my opinion. In this case especially, `hasattr` seems to make sense because I'm ultimately just looking for a single attribute anyway

Reply:
cool thanks for the link!